src/odemis/acq/move.py

```diff
@@ -368,6 +368,9 @@ def getTargetPosition(self, target_pos_lbl: int) -> Dict[str, float]:
                     sem_grid1_pos = stage_md[model.MD_SAMPLE_CENTERS][POSITION_NAMES[GRID_1]]
                     fm_target_pos = self._transformFromSEMToMeteor(sem_grid1_pos)
                 elif current_position == SEM_IMAGING:
+                    current_value = self.stage.position.value
+                    self.stage.updateMetadata({model.MD_FAV_SEM_POS_ACTIVE: {'rx': current_value['rx'],
+                                                                             'rz': current_value['rz']}})
```

Comment:
I'm not very enthusiastic in putting this code here because the function is called `getTargetPosition()`. It implies to reads/compute data, but doesn't change it. Admittedly, there are enough checks (ie, current pos is in SEM_IMAGING, and target_pos is FM_IMAGING) that it should be safe... but that doesn't feel right. 
I've looked at the `transformFromSEMToMeteor` methods: none of them use `MD_FAV_SEM_POS_ACTIVE`. So it's not really needed to update the metadata so "early" anyway. 

Hence, I think it'd be safer/clearer to place that metadata update to just before moving to the FM_IMAGING posture, inside `_doCryoSwitchSamplePosition()`. There 3 versions, but I think they all have a path `if target == FM_IMAGING:` -> `if current_label == SEM_IMAGING:`. That would go right in this block.

Reply:
I feel _doCryoSwitchSamplePosition() moves to a target position in a set order without changing the values. The transform_to_() functions do the calculation. For clarity, will it be useful to add this update metadata in the transform functions, if not in _getTargetPosition?